window-manager/src/main/java/com/codelab/foldables/window_manager/MainActivity.kt

```diff
@@ -85,48 +89,48 @@ class MainActivity : AppCompatActivity() {
         val set = ConstraintSet()
         set.clone(constraintLayout)
 
-        //We get the display feature bounds.
-        val rect = newLayoutInfo.displayFeatures[0].bounds
+        //We get the folding feature bounds.
+        val foldingFeature = newLayoutInfo.displayFeatures[0] as FoldingFeature
+        val rect = foldingFeature.bounds
 
-        //Sets the view to match the height and width of the device feature
+        //Sets the view to match the height and width of the folding feature
         set.constrainHeight(
-            R.id.device_feature,
+            R.id.folding_feature,
             rect.bottom - rect.top
         )
-        set.constrainWidth(R.id.device_feature, rect.right - rect.left)
+        set.constrainWidth(R.id.folding_feature, rect.right - rect.left)
```

Comment:
This fails when the foldingFeature as width or heigh equal to zero like on a Samsung folding device. The emulator currently has a default of one pixel width/height for the hinge and the issue is not visible there.

Reply:
Addressed in new commits. Please, let me know what you think about how it was done. Thanks!

Reply:
LGTM